### ISSUE TITLE
Install essential tools to all images generated by dazzle v2

### ISSUE THREAD

**Issue by princerachit** [gitpod-io/workspace-images#595]
We are in the process of moving image creation to dazzlev2. With this change we want to get rid of creating humongous `full` image and using it as a base for other images.

Since a lot of images are already based out of `full` image, customers might have developed dependency on some of these tools and would expect them to be present when they migrate to new images.

We need to find out the essential tools that must be shipped with all images e.g. nettools, docker etc.

**Comment by princerachit** [comment]
We have introduced docker and tailscale as essential tools/chunks for now

**Comment by atduarte** [comment]
Related https://github.com/gitpod-io/workspace-images/issues/675

**Comment by apolopena** [comment]
Can you please let people know how they can keep their existing integrations from breaking when you strip out elements that are depended on such as Apache2.

A hotfix like this took me quite a bit of testing to figure out.
https://github.com/apolopena/gitpod-laravel-starter/commit/d4dc2f6a98e442fab945bf586f1395c42324b410

If breaking changes are introduced then workarounds should be provided when those breaking changes are made.
Thanks.

**Comment by atduarte** [comment]
@apolopena you are right. Having most workspaces pointing to mutable images (i.e. `latest`) makes it hard for us to avoid introducing breaking changes. We will try to handle this better in the future.

Regarding `legacy-dazzle-v1`: this points to a older version that for example has `apache2` (https://github.com/gitpod-io/workspace-images/issues/675) but at the same time also has the older ca-certificates that also breaks other workspaces (https://github.com/gitpod-io/gitpod/issues/8059).

My recommendation would be to use specific tags such as `gitpod/workspace-mysql:2022-02-04-10-54-10` to ensure immutability.

**Comment by apolopena** [comment]
@atduarte I understand. Thank you for the context and recommendation. I appreciate that. I do have a few questions:

How can I find these tags I can use to ensure immutablility?

In regard to `workspace-full`, doesn't the name indicate that this would be the Gitpod image that allows a "full" workspace that would always contain certain elements such as apache2, nginx and PHP? How slim to you intend to make `workspace-full` in the future? Are you taking apache2, php nginx out? Wouldn't this new trim version of `workspace-full` be better named `workspace-slim` or something like that? 
I was using gitpods mutable workspace for mysql so I could get the changes Gitpod would make to it moving forward, I went through extensive code changes to avoid pinning the version of `workspace-mysql`. I guess I assumed that the changes Gitpod would make would be mostly version bumps and some additional libraries. Removing core items from something named `-full` I did not expect.

**Comment by atduarte** [comment]
> How can I find these tags I can use to ensure immutablility?

https://hub.docker.com/r/gitpod/workspace-full/tags

> In regard to workspace-full

`workspace-full` actually has apache2, nginx and php (https://github.com/gitpod-io/workspace-images/blob/master/dazzle.yaml#L31). Sorry for creating the confusion in my earlier comment, I was referring to `workspace-mysql´ as an example of the problem.

**Comment by apolopena** [comment]
Using `gitpod/workspace-mysql:2022-02-04-10-54-10` does not seem to give me apache either.

**Comment by atduarte** [comment]
Timestamped tags, such as `gitpod/workspace-mysql:2022-02-04-10-54-10`, and the `gitpod/workspace-mysql:latest` image are `gitpod/workspace-base` with only mysql installed. This is what originated https://github.com/gitpod-io/workspace-images/issues/675 and that we are discussing internally.

Currently, `gitpod/workspace-mysql:legacy-dazzle-v1` is the only one that is based on `gitpod/workspace-full` and therefore has apache and php. If you prefer to use this one for the time being, you can solve the ca-certificates problem by creating a custom Dockerfile ( `.gitpod.Dockerfile`) like the following:
```
FROM gitpod/workspace-mysql:legacy-dazzle-v1

RUN sudo apt-get update && sudo apt-get install ca-certificates
```
And adding this to the `.gitpod.yml`:
```
image:
  file: .gitpod.Dockerfile
```

For more information on how to use a custom Dockerfile please check this [page](https://www.gitpod.io/docs/config-docker#configure-a-custom-dockerfile).
